packages/client-core/src/transports/SocketWebRTCClientFunctions.ts

```diff
@@ -146,7 +141,7 @@ const handleFailedConnection = (locationConnectionFailed) => {
       !locationInstanceConnectionState.instances[instanceId]?.connected?.value &&
       !locationInstanceConnectionState.instances[instanceId]?.connecting?.value
     ) {
-      dispatchAction(LocationInstanceConnectionAction.disconnect({ instanceId }))
+      getMutableState(LocationInstanceState).instances[instanceId].set(none)
```

Comment:
everything to do with mutating `LocationInstanceState`, `MediaInstanceState` in this file should be moved to helper functions in their respective state service files. this will make this code much easier to refactor in the future.